test/index.js

```diff
@@ -465,1355 +347,959 @@ describe('Schwifty', () => {
-
-                    expect(err).to.not.exist();
-
-                    const plugin = (srv, opts, next) => {
+                const plugin = {
+                    name: 'plugin',
+                    register: (srv, opts) => {
 
                         srv.schwifty(TestModels.Person);
-                        next();
-                    };
-
-                    plugin.attributes = { name: 'plugin' };
-
-                    server.register(plugin, (err) => {
-
-                        expect(err).to.not.exist();
-
-                        server.initialize((err) => {
-
-                            expect(err).to.exist();
-                            expect(err.message).to.startWith('Could not connect to database using schwifty knex instance for models: "Dog", "Person".');
+                    }
+                };
 
-                            done();
-                        });
-                    });
-                });
+                await server.register(plugin);
+                await expect(server.initialize()).to.reject(/^Could not connect to database using schwifty knex instance for models: "Dog", "Person"\./);
             });
 
-            it('and doesn\'t list associated models in error when there are none.', (done) => {
+            it('and doesn\'t list associated models in error when there are none.', async () => {
 
                 const knex = failKnexWith(makeKnex(), new Error());
+                const server = await getServer({ knex });
 
-                getServer({ knex }, (err, server) => {
-
-                    expect(err).to.not.exist();
-
-                    server.initialize((err) => {
-
-                        expect(err).to.exist();
-                        expect(err.message).to.startWith('Could not connect to database using schwifty knex instance.');
-
-                        done();
-                    });
-                });
+                await expect(server.initialize()).to.reject(/^Could not connect to database using schwifty knex instance\./);
             });
 
-            it('and augments the original error\'s message.', (done) => {
+            it('and augments the original error\'s message.', async () => {
 
                 const error = new Error('Also this other thing went wrong.');
                 const knex = failKnexWith(makeKnex(), error);
+                const server = await getServer({ knex });
 
-                getServer({ knex }, (err, server) => {
-
-                    expect(err).to.not.exist();
-
-                    server.initialize((err) => {
-
-                        expect(err).to.exist();
-                        expect(err).to.shallow.equal(error);
-                        expect(err.message).to.equal('Could not connect to database using schwifty knex instance.: Also this other thing went wrong.');
-
-                        done();
-                    });
-                });
+                const thrown = await expect(server.initialize()).to.reject();
```

Comment:
oh ill! I didn't know reject() worked like this. very cool pattern 🙏

Reply:
And it's not just about cleanup– this is mostly for the integrity of the test.  What if `await server.initialize()` didn't throw and as a result the assertions in the catch block never ran?